### PR TITLE
docs: Update postgres aurora info

### PR DESCRIPTION
**Pull request by tara-det-ai**
Update instances of "postgresql <14", add note about Aurora v1

### CODE REVIEW

File: docs/setup-cluster/checklists/postgresql.rst [hunk 1]
```diff
@@ -75,21 +75,21 @@ Determined uses a PostgreSQL database to store experiment and trial metadata.
...
+#. Install PostgreSQL 14 or later.
 
    **Debian Distributions**
 
    On Debian distributions, use the following command:
 
    .. code::
 
-      sudo apt install postgresql-10
+      sudo apt install postgresql-14
```
Comment (rb-determined-ai) [inline docs/setup-cluster/checklists/postgresql.rst:86]: I'd just do:
```
sudo apt install postresql
```

Which will be new enough on any recent debian version.

File: docs/setup-cluster/checklists/postgresql.rst [hunk 2]
```diff
@@ -75,21 +75,21 @@ Determined uses a PostgreSQL database to store experiment and trial metadata.
 Install PostgreSQL using ``apt`` or ``yum``
 ===========================================
 
-#. Install PostgreSQL 10 or greater.
+#. Install PostgreSQL 14 or later.
```
Comment (rb-determined-ai) [inline docs/setup-cluster/checklists/postgresql.rst:78]: I'd say "Install PostgreSQL."

**Review by rb-determined-ai — COMMENTED** [review]
I talked to @stoksc, and it sounds like our official stance is that we support any non-EOL version of postgres.

So it is ok to say we _recommend_ version 14 or later, but we can't say you _need_ version 14 or later.

Personally, I think support EOL software is never implied, and we can mostly just say "Install Postgres" everywhere and people should know that EOL postgres isn't gonna work.

If you want to specify that we don't support non-EOL postgres, I'd do it only once, maybe in the `setup-cluster/checklists/postgresql.rst` file, or a file of equal specificity.

I would not tell users they should install `postgresql-14` on debian, for instance; I think they should take whatever postgresql version is default for their linux installation (the unmarked `postgresql` package).

Does that all make sense?

**Comment by ioga** [comment]
>Personally, I think support EOL software is never implied, and we can mostly just say "Install Postgres" everywhere and people should know that EOL postgres isn't gonna work.

I'd suggest we specify the minimum required version, since there's a slice of customers who run older systems, and CEs always end up asking us what's the minimum.

**Comment by stoksc** [comment]
Elsewhere I'd just recommend the newest Postgres version available.

**Comment by tara-det-ai** [comment]
> We should at least say the minimum [here](https://github.com/determined-ai/determined/pull/10116/files#diff-2320b9170fc89a2ab01ebdf1eb8c4e6f364df1b33ea4b6d22748a2fd470563bfR112) still.

Keeping "The solution requires PostgreSQL 14 or later, which will be installed on the admin node." in hpc-environment-requirements.rst file

File: docs/setup-cluster/slurm/hpc-environment-requirements.rst [hunk 3]
```diff
@@ -109,7 +109,7 @@ The following software components are required:
 Database Requirements
 =====================
 
-The solution requires PostgreSQL 10 or newer, which will be installed on the admin node. The
+The solution requires PostgreSQL 14 or later, which will be installed on the admin node. The
```
Comment (rb-determined-ai) [inline docs/setup-cluster/slurm/hpc-environment-requirements.rst:112]: Minimum required version is 13 I think.  Or, it's 12, for like two more weeks.

**Review by rb-determined-ai — APPROVED** [review]

**Pull request merged**